test/end-to-end-tests/src/rest/room.ts

```diff
@@ -19,25 +19,30 @@ import uuidv4 = require('uuid/v4');
 
 import { RestSession } from "./session";
 import { Logger } from "../logger";
+import { RestThread } from "./thread";
 
-/* no pun intented */
+/* no pun intended */
 export class RestRoom {
     constructor(readonly session: RestSession, readonly roomId: string, readonly log: Logger) {}
 
-    async talk(message: string): Promise<void> {
+    async talk(message: string): Promise<string> {
         this.log.step(`says "${message}" in ${this.roomId}`);
         const txId = uuidv4();
-        await this.session.put(`/rooms/${this.roomId}/send/m.room.message/${txId}`, {
+        const { event_id: eventId } = await this.session.put(`/rooms/${this.roomId}/send/m.room.message/${txId}`, {
             "msgtype": "m.text",
             "body": message,
         });
         this.log.done();
-        return txId;
+        return eventId;
     }
 
     async leave(): Promise<void> {
         this.log.step(`leaves ${this.roomId}`);
         await this.session.post(`/rooms/${this.roomId}/leave`);
         this.log.done();
     }
+
+    thread(threadId: string): RestThread {
```

Comment:
public/private, though noting that the rest of the file doesn't seem to care - fixing them would be optional in this PR.